internal/config/substitutions_test.go

```diff
@@ -68,3 +69,47 @@ func TestSubTimestamps(t *testing.T) {
 
 	assert.Equal(t, expected, fileContent)
 }
+
+func Test_parseLookupFileAndUpdateConfig(t *testing.T) {
+
+	testConfig := `
+custom_attributes:
+  replace_id_float: ${lf:id_float}
+  replace_id_int: ${lf:id_int}
+  replace_name: ${lf:name}
+`
+
+	item := map[string]interface{}{
+		"id_float": 2456853.0,
+		"id_int":   2456854,
+		"name":     "AMP_eov_ntet-np",
+	}
+
+	actual, err := fillTemplateConfigWithValues(item, testConfig)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, actual)
+
+	expected := load.Config{
+		CustomAttributes: map[string]string{
+			"replace_id_float": "2456853.000000",
+			"replace_id_int":   "2456854",
+			"replace_name":     "AMP_eov_ntet-np",
+		},
+	}
+
+	assert.ObjectsAreEqual(expected.CustomAttributes, (*actual).CustomAttributes)
+}
+
+func Test_getSprintfTemplateForVariable(t *testing.T) {
+	var valueInt int = 2456853
+	assert.Equal(t, "%d", getSprintfTemplateForVariable(valueInt))
+	var valueFloat32 float32 = 2456853
+	assert.Equal(t, "%f", getSprintfTemplateForVariable(valueFloat32))
+	var valueFloat64 float64 = 2456853
+	assert.Equal(t, "%f", getSprintfTemplateForVariable(valueFloat64))
+	var valueString string = "2456853"
```

Comment:
You can use also use short variable declaration:
       `valueString := "2456853"
         valueInt := 2456853
         valueFloat64 := float64(2456853)
         valueFloat32 := float32(2456853)` 

Go will infer the data type when is string or int. For float you can use the type builtin

Reply:
Yeah, I was thinking that way, but then turned it into explicit type declaration. Reason is that it will be more visible
- int and we expect %d, 
- string and %s, etc. 
Not sure is it worth it or not.